Mixmatch/streetview_dataset/parse_data_to_tfrecord_lib.py

```diff
@@ -0,0 +1,207 @@
+    xmin = tf.sparse_tensor_to_dense(image_features[BBOX_XMIN], default_value=0).numpy()
+    ymin = tf.sparse_tensor_to_dense(image_features[BBOX_YMIN], default_value=0).numpy()
+    xmax = tf.sparse_tensor_to_dense(image_features[BBOX_XMAX], default_value=0).numpy()
+    ymax = tf.sparse_tensor_to_dense(image_features[BBOX_YMAX], default_value=0).numpy()
+    
+    bbox = np.vstack((xmin, ymin, xmax, ymax)) # Left, Top, Right, Bottom
+    
+    return img_name, confidence, bbox
+
+
+# Transform raw image data and label into a tfexample format.
+def img_to_example(img, label):
+    imgByteArr = io.BytesIO()
+    img.save(imgByteArr, format='JPEG')
+    imgByteArr = imgByteArr.getvalue()
+
+    example = tf.train.Example(features=tf.train.Features(feature={
+        "image": tf.train.Feature(bytes_list=tf.train.BytesList(value=[imgByteArr])),
+        "label": tf.train.Feature(int64_list=tf.train.Int64List(value=[label]))}))
+    
+    return example
+
+
+# Write all images into the test TFrecord file.
+def write_tfrecord_from_images(image_folder_path, label, writer):
+    for img_name in os.listdir(image_folder_path):
+        img_path = os.path.join(image_folder_path, img_name)
+        
+        try:
+            img = Image.open(img_path, "r")
+        except Exception as e:
+            print(e)
+            print(img_path + " is not valid")
+            continue
+            
+        # Exclude all non RGB images
+        if len(img.getbands()) != 3:
+            continue
+
+        img = img.resize(OUTPUT_IMAGE_SIZE)
+
+        example = img_to_example(img, label)
+        writer.write(example.SerializeToString())
+
+# Striped out only the maximum confidence bbox of a image. Function is called in generate_tfexamples_from_detections().
+def strip_top_confidence_bbox(confidence, bbox):
+    target = []
+            
+    if confidence.size > 0 and max(confidence) > CONF_THRESHOLD['pos']:
+        pos = np.argmax(confidence)
+        target.append({'label':1, 'bbox':bbox[:, pos]})
+    elif confidence.size == 0 or (confidence.size > 0 and max(confidence) < CONF_THRESHOLD['neg']):
+        target.append({'label':0})
+    
+    return target
+
+
+# Striped out ALL bbox where confidence is over threshold. Function is called in generate_tfexamples_from_detections().
+def strip_all_qualified_bbox(confidence, bbox):
+    target = []
+    if confidence.size > 0:
+        for i in range(confidence.size):
+            if confidence[i] > CONF_THRESHOLD['pos'] or confidence[i] < CONF_THRESHOLD['neg']:
+                target.append({'label':int(round(confidence[i])), 'bbox':bbox[:, i]})
+    
+    return target
+
+
+# Strip the bboxes from the parsed_image_dataset that are over threshold and write it into TFrecord file.
+def generate_tfexamples_from_detections(parsed_image_dataset, folder_path, include_top_camera, only_keep_top_confidence):
+    examples = {0:[], 1:[]}
```

Comment:
Describe this dict, i.e. 0 for negative examples and 1 for positive examples.

Reply:
Done